### PR TITLE
[SELC-5791] Feat: change name for prod-interop and refactor modal logic

### PR DESCRIPTION
**Pull request by OraldoDoci**
#### List of Changes
Show only prod-interop title when authorized only on one enviroment
Refactor and simplify code.

#### Motivation and Context
Request by design to only show the name of the prod-interop product

#### How Has This Been Tested?
Tested by running application locally

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/pages/dashboardOverview/components/activeProductsSection/ActiveProductsSection.tsx [hunk 1]
```diff
@@ -1,94 +1,80 @@
...
+type Props = {
+  party: Party;
+  products: Array<Product>;
+};
+
+export default function ActiveProductsSection({ party, products }: Readonly<Props>) {
+  const { t } = useTranslation();
+  const { hasPermission } = usePermissions();
+
+  const interopProductIds = ['prod-interop', 'prod-interop-atst', 'prod-interop-coll'];
```
Comment (manuraf) [inline src/pages/dashboardOverview/components/activeProductsSection/ActiveProductsSection.tsx:20]: do we have enum or constants about these values?

File: src/pages/dashboardOverview/components/activeProductsSection/ActiveProductsSection.tsx [hunk 2]
```diff
@@ -1,94 +1,80 @@
...
+        mbTitle={2}
+        variantTitle="h5"
+        titleFontSize="22px"
+      />
+      <Grid container spacing={3}>
+        {party.products
+          .filter(
+            (us) =>
+              us.productOnBoardingStatus === 'ACTIVE' &&
+              (us.productId?.startsWith('prod-interop') && hasMoreThanOneInteropEnv
```
Comment (manuraf) [inline src/pages/dashboardOverview/components/activeProductsSection/ActiveProductsSection.tsx:48]: ditto

File: src/pages/dashboardOverview/components/activeProductsSection/components/ActiveProductCardContainer.tsx [hunk 3]
```diff
@@ -51,20 +47,27 @@ export default function ActiveProductCardContainer({
 
   const lang = i18n.language;
 
+  const startWithProductInterop = (id?: string) =>
+    id !== undefined && id.startsWith('prod-interop');
```
Comment (manuraf) [inline src/pages/dashboardOverview/components/activeProductsSection/components/ActiveProductCardContainer.tsx:51]: ditto

File: src/pages/dashboardOverview/components/activeProductsSection/ActiveProductsSection.tsx (hunk 1 above)
Comment (OraldoDoci) [reply src/pages/dashboardOverview/components/activeProductsSection/ActiveProductsSection.tsx]: Yes

**Review by manuraf — APPROVED** [review]

**Pull request merged**